realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -307,6 +307,34 @@ private boolean categorizeClassElements() {
                     }
                 }
 
+                if (variableElement.getAnnotation(Required.class) == null) {
+                    // The field doesn't have the @Required annotation.
+                    // Without @Required annotation, boxed types/RealmObject/Date/String/bytes should be added to
+                    // nullableFields.
+                    // RealmList and Primitive types are NOT nullable always. @Required annotation is not supported.
+                    if (!Utils.isPrimitiveType(variableElement) && !Utils.isRealmList(variableElement)) {
+                        nullableFields.add(variableElement);
+                    }
+                } else {
+                    // The field has the @Required annotation
+                    if (Utils.isPrimitiveType(variableElement)) {
+                        Utils.error("@Required is not needed for field " + element +
+                                " with the type " + element.asType());
+                    } else if (Utils.isRealmList(variableElement)) {
+                        Utils.error("@Required is invalid for field " + element +
+                                " with the type " + element.asType());
+                    } else if (Utils.isRealmObject(variableElement)) {
+                        Utils.error("@Required is invalid for field " + element +
+                                " with the type " + element.asType());
+                    } else {
+                        // Should never get here - user should remove @Required
+                        if (nullableFields.contains(variableElement)) {
+                            Utils.error("Annotated field " + element + " with type " + element.asType() +
```

Comment:
this is most likely an annotation error, user should clean (sometimes [kill instances of Gradle](https://discuss.gradle.org/t/gradle-not-compiles-with-solder-tooling-jar/7583/21)) then try again,  removing @Required will create more confusion